sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DDLParser.scala

```diff
@@ -173,13 +174,15 @@ class DDLParser(parseQuery: String => LogicalPlan)
     optionName ~ stringLit ^^ { case k ~ v => (k, v) }
 
   protected lazy val column: Parser[StructField] =
-    ident ~ dataType ~ (COMMENT ~> stringLit).?  ^^ { case columnName ~ typ ~ cm =>
-      val meta = cm match {
-        case Some(comment) =>
-          new MetadataBuilder().putString(COMMENT.str.toLowerCase, comment).build()
-        case None => Metadata.empty
-      }
-
-      StructField(columnName, typ, nullable = true, meta)
+    ident ~ dataType ~ (NOT ~ NULL).? ~ (COMMENT ~> stringLit).?  ^^ {
+      case columnName ~ typ ~ setNotNullable ~ cm =>
```

Comment:
I guess that makes it a little less ambiguous. I have made the change in the latest commit.